src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -0,0 +1,120 @@
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseException;
+import com.google.gson.Gson;
+import com.google.step.finscholar.data.ServletConstantValues;
+import java.util.logging.Logger;
+
+/** This class handles writing to and reading from a Cloud Firestore database. */
+public class FirebaseStorageManager {
+  /** This is used to convert Objects into JSON strings to send to the frontend. */
+  private static Gson gson = new Gson();
+
+  /** Formatters to use with String.format(). */
+  private static final String ADDED_NEW_DOC_FORMATTER = "New document added to %s at %s.";
+  private static final String EXCEPTION_DNE_FORMATTER = "Document with id: %s does not exist.";
+  private static final String EXCEPTION_FORMATTER = "%s%s";
+
+  /** Logger that sends logs to the Cloud Project console. */
+  private static final Logger log = Logger.getLogger(FirebaseStorageManager.class.getName());
+
+  /** 
+   * This method stores an object as a DocumentReference (a data point) 
+   *   inside a Collection (of DocumentReferences) in Firestore. 
+   * @param database - This is the database I want to write my data point to.
+   * @param collectionToWriteTo - This is the collection of Documents I want to write to.
+   * @param object - The object that will be stored as a new Document in the collectionToWriteTo.
+   *   It doesn't matter what type the object is as DocumentReference.set() is type-agnostic and will
+   *   store any serializable object. This object is required to have a public, no-parameter constructor.
+   *   Also, the instance variables for that object should not be declared as final. 
+   *   There are no other requirements for the object.
+   * @param documentID - The optional id for storing the document.
+   */
+  public static void storeDocument(Firestore database, String collectionToWriteTo, Object object, String documentID) 
+    throws FirebaseException {
+    // Access the correct collection.
+    CollectionReference collectionRef = database.collection(collectionToWriteTo);
+
+    // Access/create the new document.
+    DocumentReference documentRef;
+    if (documentID.equals(ServletConstantValues.DEFAULT_VALUE)) {
+      documentRef = collectionRef.document();
+    } else {
+      documentRef = collectionRef.document(documentID);
+    }
+
+    // Update the document with a new object.
+    ApiFuture<WriteResult> future = documentRef.set(object);
+    try {
+      // Log that a new document has been added to database.
+      String message = String.format(ADDED_NEW_DOC_FORMATTER, 
+          collectionToWriteTo, future.get().getUpdateTime().toString());
+      log.info(message);
+
+    } catch (Exception e) {
+      // Throws a FirebaseException if unsuccessful in adding new document.
+      String message = String.format(EXCEPTION_FORMATTER, 
+        ServletConstantValues.UNABLE_TO_WRITE_TO_FIRESTORE , 
```

Comment:
Does this still need to be a constant or can you change it like you did the other formatters? The formatable string itself could also be the constant if you need it elsewhere.